x-pack/lib/config_management/elasticsearch_source.rb

```diff
@@ -63,33 +77,21 @@ def pipeline_configs
           end
         end
 
-        response = fetch_config(pipeline_ids)
-
-        if response["error"]
-          raise RemoteConfigError, "Cannot find find configuration for pipeline_id: #{pipeline_ids}, server returned status: `#{response["status"]}`, message: `#{response["error"]}`"
-        end
-
-        if response["docs"].nil?
-          logger.debug("Server returned an unknown or malformed document structure", :response => response)
-          raise RemoteConfigError, "Elasticsearch returned an unknown or malformed document structure"
-        end
+        fetcher = get_pipeline_fetcher
+        response = fetcher.fetch_config(pipeline_ids, client)
 
-        # Cache pipelines to handle the case where a remote configuration error can render a pipeline unusable
-        # it is not reloadable
-        @cached_pipelines = response["docs"].collect do |document|
-          get_pipeline(document)
+        @cached_pipelines = pipeline_ids.collect do |pid|
+          get_pipeline(pid, response, fetcher)
         end.compact
       end
 
-      def get_pipeline(response)
-        pipeline_id = response["_id"]
-
-        if response["found"] == false
+      def get_pipeline(pipeline_id, response, fetcher)
+        if response.has_key?(pipeline_id) == false
```

Comment:
Nit: Consider using `unless` instead of `if X == false`, eg `unless response.has_key?(pipeline_id)`